### PR TITLE
Fixed issue with PicklingError causes the timeout to never be reached.

### PR DESCRIPTION
**Pull request by hvdklauw**

### CODE REVIEW

File: timeout_decorator/timeout_decorator.py [hunk 1]
```diff
@@ -135,17 +134,15 @@ def cancel(self):
...
     def ready(self):
         """Read-only property indicating status of "value" property."""
-        if self.__queue.full():
-            return True
-        elif not self.__queue.empty():
-            return True
-        elif self.__timeout < time.time():
+        if self.__timeout < time.time():
             self.cancel()
+        if self.__queue.full() and not self.__queue.empty():
```
Comment (pnpnpn) [inline timeout_decorator/timeout_decorator.py]: I think this can be just `return self.__queue.full() and not self.__queue.empty()`

**Comment by pnpnpn** [comment]
@bubenkoff

File: CHANGES.rst [hunk 2]
```diff
@@ -1,6 +1,10 @@
 Changelog
 =========
 
+0.3.1
```
Comment (pnpnpn) [inline CHANGES.rst]: You need to bump the version setup.py

**Comment by hvdklauw** [comment]
I actually debugged this issue together with @bubenkoff :)

Also, the full and not empty might seem strange, but we discovered that the meaning in multiprocessing is actually not obvious , in this particular case the queue would report both empty and full (The process is in the queue so it's full, but the results are empty because the process got a PickleError.)

The result would be that the self.__queue.get() would hang forever waiting to get a result that never comes.

**Comment by bubenkoff** [comment]
@pnpnpn we've found it out during the field experiments with mocking :)
It was a hard time but now it works correctly even if multiprocessing fails in itself

**Comment by pnpnpn** [comment]
:+1:

**Pull request merged**